From 1b75213d7ff5ff9fc17fa6504667f2b6bf52d2c2 Mon Sep 17 00:00:00 2001
From: Elon Zhang <zhangzj@rock-chips.com>
Date: Tue, 25 Jun 2019 11:53:06 +0800
Subject: [PATCH] OP-TEE: fix IS_ERR_VALUE compilation error

Fix the compilation error below:
error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]

Change-Id: Ib7cbc813c6eeeded220171ef5e4434ad798d06d9
Signed-off-by: Elon Zhang <zhangzj@rock-chips.com>
---
 security/optee_linuxdriver/core/tee_shm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/optee_linuxdriver/core/tee_shm.c b/security/optee_linuxdriver/core/tee_shm.c
index 4b09789ddceb..f5e5c2e904ce 100644
--- a/security/optee_linuxdriver/core/tee_shm.c
+++ b/security/optee_linuxdriver/core/tee_shm.c
@@ -117,7 +117,7 @@ struct tee_shm *rk_tee_shm_alloc(struct tee *tee, size_t size, uint32_t flags)
 
 	ret = sg_alloc_table_from_pages(&shm->sgt, &page,
 			nr_pages, 0, nr_pages * PAGE_SIZE, GFP_KERNEL);
-	if (IS_ERR_VALUE(ret)) {
+	if (ret < 0) {
 		dev_err(_DEV(tee), "%s: sg_alloc_table_from_pages() failed\n",
 				__func__);
 		tee->ops->free(shm);
-- 
2.35.3

